openwisp_controller/config/base/vpn.py

```diff
@@ -71,9 +83,11 @@ def save(self, *args, **kwargs):
         """
         if not self.cert:
             self.cert = self._auto_create_cert()
-        if not self.dh:
-            self.dh = self.dhparam(2048)
         super().save(*args, **kwargs)
+        if not self.dh:
+            self.dh = self._placeholder_dh
```

Comment:
setting this parameter here after the save is called is causing that the object is created with an empty DH.
Your tests are not catching this bug.
Please add a test which catches this bug before fixing it, eg:

- use transaction test case
- mock the celery task so it's not effective
- assert the dh equals the placeholder

Call `super().save(*args, **kwargs)` as last line please.